src/WorkerChannel.ts

```diff
@@ -154,4 +194,54 @@ export class WorkerChannel {
       resultCallback(err);
     }
   }
+
+  /**
+   * Worker sends the host information identifying itself
+   */ 
+  public startStream(requestId: string, msg: rpc.StartStream): void {
+    // Not yet implemented
+  }
+
+  /**
+   * Message is empty by design - Will add more fields in future if needed
+   */ 
+  public workerHeartbeat(requestId: string, msg: rpc.WorkerHeartbeat): void {
+    // Not yet implemented
+  }
+
+  /**
+   * Warning before killing the process after grace_period
+   * Worker self terminates ..no response on this
+   */ 
+  public workerTerminate(requestId: string, msg: rpc.WorkerTerminate): void {
+    // Not yet implemented
+  }
+
+  /**
+   * NOT USED
+   */ 
+  public workerStatusRequest(requestId: string, msg: rpc.WorkerStatusRequest): void {

```

Comment:
Nit: This file is getting long. Move handlers to their own files

Reply:
The way the WorkerChannel is currently implemented, it looks for handlers as properties on its own object. TypeScript doesn't have a notion of partial classes, so we can't split these out